sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -559,6 +559,17 @@ object RemoveRedundantAliases extends Rule[LogicalPlan] {
         })
         Join(newLeft, newRight, joinType, newCondition, hint)
 
+      case _: Union =>
```

Comment:
Shall we update the method comment as well?

Reply:
Thanks, fixed in https://github.com/apache/spark/pull/37334/commits/db6c73f7b79ea2d7490e24390f2e9b6731ccb209